qiskit/backends/_qiskitcppsimulator.py

```diff
@@ -0,0 +1,341 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2017 IBM RESEARCH. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =============================================================================
+
+"""
+Interface to C++ quantum circuit simulator with realistic noise.
+"""
+
+import sys
+import os
+
+# Add path to external simulator
+simpath = os.path.dirname(__file__)
+simpath = os.path.join(simpath + '/../../src/cpp-simulator')
+sys.path.append(simpath)
+
+# Import C++ simualtor interface
+import qiskitsimulator
+
+# =============================================================================
+# Try loading simulator based of Master branch
+# =============================================================================
+
+try:
+    from qiskit._result import Result
+    from qiskit.backends._basebackend import BaseBackend
+    from qiskit.backends._backendutils import register_backend
+
+    class QiskitCppSimulator(BaseBackend):
+        "External C++ quantum circuit simulator with realistic noise."
+
+        def __init__(self, configuration=None):
+            if configuration is None:
+                self._configuration = qiskitsimulator.simulator_config()
+            else:
+                self._configuration = configuration
+
+        def run(self, q_job):
+            qobj = q_job.qobj
+            exe = qobj.get("config").get("exe")
```

Comment:
Where this qobj["config"]["exe"] would be filled? From some kind of user input?

Reply:
This lets user specify an alternative location for the compiled simulator binary if they move it from the default location for some reason

Reply:
Ok, so seems like we have a "path" and "exe" config parameters, wondering if there is a reason to not have just a single path parameter that points directly to the binary? like: `path = /home/user/bin/qiskit_simulator`, does it make sense?